src/components/other/other/UnauthorizedHandler.tsx

```diff
@@ -0,0 +1,21 @@
+import axios from 'axios';
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+export const UnauthorizedHandler = () => {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    axios.interceptors.response.use(
+      (response) => response,
+      (error) => {
+        const statusCode = error.response ? error.response.status : null;
+        if (statusCode === 401) {
+          navigate('/login', { state: { customMessage: 'Widzisz ten ekran, ponieważ zostałeś wylogowany' } });
```

Comment:
I think there are now multiple places where we use '/login' string in which case we should think of creating a common file with directory constants (would be much easier to change a single LOGIN constant instead of multiple /login strings throughout the code).

Reply:
Resolved